src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatReader.cs

```diff
@@ -43,6 +43,16 @@ internal object ReflectionReadClass(XmlReaderDelegator xmlReader, XmlObjectSeria
             object obj = ReflectionCreateObject(_classContract);
             context.AddNewObject(obj);
             ReflectionReadMembers(obj, xmlReader, context, memberNames, memberNamespaces);
+
+            if (obj.GetType() == typeof(DateTimeOffsetAdapter))
```

Comment:
We should not allow concrete adapters to leak into ReflectionXmlFormatReader or ReflectionXmlFormatWriter.